### ISSUE TITLE
onDisconnected not called all the time ?

### ISSUE THREAD

**Issue by amallo** [TakahikoKawasaki/nv-websocket-client#29]
When I disconnect the websocket manually  by
```javas
websocket.disconnect()
```
the onDisconnected event is not called, only the onStateChanged event is called.  

I tried to debug a little bit and  here is what I see 

- There is a call to WebSocket.stopThreads() followed by ReadingThread.requestStop() that should end the main ReadingThread loop.
- I just saw that ReadingThread.notifyFinished() is never called (in my case) but I don't understand why...

Do you have an idea ?

**Comment by amallo** [comment]
I dig a little bit more, maybe took the wrong direction but just to show you in case

in ReadingThread:

```java
private void main()
{
        ....
        while (true)
        {
            synchronized (this)
            {
                if (mStopRequested)
                {
                    break;
                }
            }

            // Receive a frame from the server.
            WebSocketFrame frame = readFrame();  <---- block here then notifyFinished() never called.
}
```

**Comment by TakahikoKawasaki** [comment]
Thank you for your report. I'll investigate the problem.

**Comment by amallo** [comment]
Hi,

have you seen something ?

**Comment by TakahikoKawasaki** [comment]
This is a bug. I guess this happens (1) if `read()` is not blocked when `disconnect()` is called and (2) if the server does not return a close frame. This issue has been in my mind since you pointed it out, and I've been thinking about how to fix it.

**Comment by TakahikoKawasaki** [comment]
@amallo I released a new version, `1.26`. I think this issue has been fixed.

**Comment by TakahikoKawasaki** [comment]
@amallo I'm closing this issue. Please let me know if this issue reproduces.
